native/packages/react-native-bpk-component-navigation-bar/src/BpkNavigationBar.android.js

```diff
@@ -181,6 +184,40 @@ class BpkNavigationBar extends Component<Props, {}> {
     let tintColor = colorWhite;
     let touchableColor = colorWhite;
 
+    let titleView = null;
+
+    // This if ensures Flow correctly refines the type of
+    // title in the body of the if to `'string' | TitleWithIcon
+    if (
+      typeof title === 'string' ||
+      (typeof title === 'object' && title.icon)
+    ) {
+      titleView = (
+        <TitleView
+          title={title}
+          tintColor={tintColor}
+          style={styles.titleString}
+        />
+      );
+    }
+
+    // This if ensures Flow correctly refines the type of
+    // title in the body of the if to `Element`.
+    // While this if is mutually exclusive to the above it
+    // cannot be an else if as Flow seems unable to handle this.
+    if (typeof title === 'object' && title.type && isValidElement(title)) {
+      titleView = (
+        <View style={styles.titleViewOuter}>
+          {React.cloneElement(title, {
+            style: [
+              title.props.style ? title.props.style : null,
+              { maxHeight: 32 },
```

Comment:
what;s the plan for the magic number? we can create a token or even al alias for it

Reply:
All of these magic numbers are intentional all though maybe these two shouldn't be. I'd argue it's out of scope for this PR though